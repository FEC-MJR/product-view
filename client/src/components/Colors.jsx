import React from 'react'
import CreateColorButton from './CreateColorButton.jsx'
import styled from 'styled-components'

const StyledLabel = styled.div`
  padding-bottom: 5px;
  color: #707070;
  font-size: 12px;
`;

const StyledButtonWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  cursor: pointer;
`;

class Colors extends React.Component {

  constructor(props) {
    super(props)

    this.state = {
      products: this.props.data.products,
      // currentProduct: this.props.data.currentProduct,
      onHover: false
    }

    this.colorHoverOn = this.colorHoverOn.bind(this)
    this.colorHoverOff = this.colorHoverOff.bind(this)
  }

  colorHoverOn(color) {
    this.setState({
      onHover: true,
      tempColor: color
    })
  }

  colorHoverOff() {
    this.setState({
      onHover: false,
      tempColor: null
    })
  }

  render() {

    const ColorButton = this.state.products.map((item, index) => (
      <CreateColorButton key={index} item={item} getCurrentProduct={this.props.getCurrentProduct} colorHoverOn={this.colorHoverOn} colorHoverOff={this.colorHoverOff}/>
    ))

    if (this.state.onHover === false) {
      return(
        <div>
          <StyledLabel>{this.props.data.currentProduct.color}</StyledLabel>
          <StyledButtonWrapper>{ColorButton}</StyledButtonWrapper>
        </div>
      )
    } else {
      return(
        <div>
          <StyledLabel>{this.state.tempColor}</StyledLabel>
          <StyledButtonWrapper>{ColorButton}</StyledButtonWrapper>
        </div>
      )
    }
  }
}

export default Colors