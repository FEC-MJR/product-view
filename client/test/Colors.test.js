import React from 'react'
import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer'
import Colors from '../src/components/Colors.jsx'

describe("Component testing for <Colors/>", () => {

  const props = {
    data:  {
      products: [{
        name: "KANKEN LAPTOP 13\"",
        type: "backpack",
        description: "Sed eu eros. Nam consequat dolor vitae dolor. Donec fringilla. Donec feugiat metus sit",
        userRating: 5,
        totalRatings: 6,
        price: 110.00,
        gender: "Unisex",
        style: "Style # F27171",
        size: ["One Size"],
        color: "166 - Acorn",
        image: ["https://hrphxfec.s3-us-west-1.amazonaws.com/Kanken+Laptop+13/kanken_acorn.jpg"]
      }],
      currentProduct : {
        name: "KANKEN LAPTOP 13\"",
        type: "backpack",
        description: "Sed eu eros. Nam consequat dolor vitae dolor. Donec fringilla. Donec feugiat metus sit",
        userRating: 5,
        totalRatings: 6,
        price: 110.00,
        gender: "Unisex",
        style: "Style # F27171",
        size: ["One Size"],
        color: "166 - Acorn",
        image: ["https://hrphxfec.s3-us-west-1.amazonaws.com/Kanken+Laptop+13/kanken_acorn.jpg"]
      }
    }
  }
  
  test("<Colors/> should render", () => {

    const wrapper = shallow(<Colors {...props}/>)
    expect(wrapper)
  })

  test("<Colors/> snapshot", () => {

    const tree = renderer.create(<Colors {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})