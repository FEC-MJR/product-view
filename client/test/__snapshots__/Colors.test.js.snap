// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing for <Colors/> <Colors/> snapshot 1`] = `
<div>
  <div
    className="sc-AxiKw hKBFOs"
  >
    166 - Acorn
  </div>
  <div
    className="sc-AxhCb jAlFeM"
  >
    <span
      className="sc-AxirZ hEOhER"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        className="sc-AxjAm jWIhmU colored-button"
        data="166 - Acorn"
      />
    </span>
  </div>
</div>
`;
